Test/Invoke-WhiskeyPester3Task.Tests.ps1

```diff
@@ -6,13 +6,144 @@ Set-StrictMode -Version 'Latest'
 . (Join-Path -Path $PSScriptRoot -ChildPath '..\Whiskey\Functions\Publish-WhiskeyPesterTestResult.ps1' -Resolve)
 
 $context = $null
+$pesterPath = $null
+$version = $null
+$taskParameter = @{}
+$failed = $false
 
-function Assert-PesterRan
+function GivenTestContext
+{
+    $script:pesterPath = $null
+    $script:version = $null
+    $script:failed = $false
+    $script:taskParameter = @{}
+    $Global:Error.Clear()
+    $script:context = New-WhiskeyPesterTestContext
+}
+
+function New-WhiskeyPesterTestContext 
+{
+    param()
+    process
+    {
+        $outputRoot = Join-Path -Path $TestDrive.FullName -ChildPath '.\.output'
+        if( -not (Test-Path -Path $outputRoot -PathType Container) )
+        {
+            New-Item -Path $outputRoot -ItemType 'Directory' | Out-Null
+        }
+        $buildRoot = Join-Path -Path $PSScriptRoot -ChildPath 'Pester' -Resolve
+        $script:context = New-WhiskeyTestContext -ForTaskName 'Pester3' -ForOutputDirectory $outputRoot -ForBuildRoot $buildRoot -ForDeveloper
+        return $context
+    }
+}
+function GivenVersion
 {
     param(
         [string]
-        $ReportsIn,
+        $Version
+    )
+    $Script:taskparameter['version'] = $Version
+}
+function GivenInvalidVersion
+{
+    $Script:taskparameter['version'] = '3.0.0'
+    Mock -CommandName 'Test-Path' `
+        -ModuleName 'Whiskey' `
+        -MockWith { return $False }`
+        -ParameterFilter { $Path -eq $context.BuildRoot }
+}
+function GivenPesterPath
+{
+    param(
+        [string[]]
+        $pesterPath
+    )
+    $script:taskParameter['path'] = $pesterPath 
+}
+
+function GivenFindModuleFails
+{
+    Mock -CommandName 'Find-Module' -ModuleName 'Whiskey' -MockWith { return $Null }
+    Mock -CommandName 'Where-Object' -ModuleName 'Whiskey' -MockWith { return $Null }
+}
+function GivenWithCleanFlag
+{
+    $context.RunMode = 'Clean'
+    Mock -CommandName 'Uninstall-WhiskeyTool' -ModuleName 'Whiskey' -MockWith { return $true }
+}
+function GivenWithInitilizeFlag
+{
+    $context.RunMode = 'initialize'
+}
+function WhenPesterTaskIsInvoked
+{
+    [CmdletBinding()]
+    param(
+        [Switch]
+        $WithClean
+    )
+
+    $failed = $false
+    $Global:Error.Clear()
 
+    Mock -CommandName 'Publish-WhiskeyPesterTestResult' -ModuleName 'Whiskey'
+
+    try
+    {
+        Invoke-WhiskeyPester3Task -TaskContext $context -TaskParameter $taskParameter

```

Comment:
Change to use `Invoke-WhiskeyTask` so the test verifies that the task's `ShouldInitialize` property is `$true`. Ditto Pester4 test/task.

Reply:
Done for all